src/Compiler/Driver/fsc.fs

```diff
@@ -887,6 +887,38 @@ let main3
             for f in checkedImplFileAfterOptimizations do
                 TailCallChecks.CheckImplFile(tcGlobals, tcImports.GetImportMap(), true, f.ImplFile.Contents)
 
+    let refAssemblySignatureHash =
```

Comment:
Was there any inspiration for this? I mean, I don't know much about signature hashing, just wondering what was the thinking behind the algorithm (maybe it's basic).

Reply:
@psfinaki mmh, I'm confused. The hashing is from the merge of main into this branch.

Reply:
Maybe it's worth rebasing to `main`, so the actual diff is easier to seen? 🙂

Reply:
Yeah please rebase then, I got confused also, sorry :D